src/App.tsx

```diff
@@ -1,96 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+// import { createBrowserRouter } from 'react-router-dom';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { ItemList } from './components/ItemList';
+import { PerPageList } from './components/PerPageList';
+
+const defaultPaginVal = {
+  total: 42,
+  perPage: 5,
+  currentPage: 1,
+};
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const options = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [paginationOption, setPaginationOption] = useState({
+    total: defaultPaginVal.total,
+    perPage: defaultPaginVal.perPage,
+    currentPage: defaultPaginVal.currentPage,
+  });
+
+  const itemPerPageHandler = (value: string) => setPaginationOption({
+    ...paginationOption,
+    ...{ perPage: +value, currentPage: defaultPaginVal.currentPage },
+  });
+
+  const currentPageHandler = (value: number) => setPaginationOption({
+    ...paginationOption, ...{ currentPage: value },
+  });
```

Comment:
If we want to work with the actual state, it should be received through a callback, and not applied directly.
It is also not necessary to wrap the field in an object and immediately open it if it is not necessary, just overwrite the required fields:
```js
const oldObject = {
  field1: 1,
  field2: 2
};
const newObject = {
  ...oldObject,
  field1: 2
};
const object = {
  ...newObject,
  field2: 1
};
```